Sources/Moya/TargetType.swift

```diff
@@ -38,32 +32,36 @@ public extension TargetType {
     }
 }
 
-/// Represents a type of upload task.
-public enum UploadType {
+/// Represents an HTTP task.
+public enum Task {
 
-    /// Upload a file.
-    case file(URL)
+    /// A request with no additional data.
+    case requestPlain
 
-    /// Upload "multipart/form-data"
-    case multipart([MultipartFormData])
-}
+    /// A requests body set with data.
+    case requestData(Data)
 
-/// Represents a type of download task.
-public enum DownloadType {
+    /// A requests body set with encoded parameters.
+    case requestParameters(parameters: [String: Any], encoding: ParameterEncoding)
 
-    /// Download a file to a destination.
-    case request(DownloadDestination)
-}
+    /// A requests body set with data, combined with url parameters.
+    case requestCompositeData(bodyData: Data, urlParameters: [String: Any])
 
-/// Represents an HTTP task.
-public enum Task {
+    /// A requests body set with encoded parameters combined with url parameters.
+    case requestCompositeParameters(bodyParameters: [String: Any], bodyEncoding: ParameterEncoding, urlParameters: [String: Any])
```

Comment:
I'm actually against removing the case only because of one possible way of misusing it. I agree though that we should deal with that situation somehow to minimize any unexpected behavior: For that I think throwing a fatalError would be possible. What do you think @sunshinejr? @SD10 seemed to be okay with that.